### DISCUSSION TITLE
Redmi Note 7 and Dexcom G6: Connection randomly drops

### DISCUSSION THREAD

**Discussion by dagord** [NightscoutFoundation/xDrip#2167] [Q&A]
Hello,
while using XDrip+ for a few days, I'm experiencing a very unstable connection between my phone (Xiaomi Redmi Note 7, Android 10 QKQ1, no custom roms) and my Dexcom G6 sensor.
Every now and then the connection drops, even if I have the phone at my side, then it reconnects automatically after a random time (it could be minutes or hours).
The sensor was started with the "unofficial" Dexcom app, which worked flawlessly in the last two months; then I removed it in order to test XDrip. So in the meanwhile I had to use the Dexcom standalone reader.
Eventually, I decided to switch back to the Dexcom app, and it still works, so it's an issue related to XDrip.
The logs are verbose, but I can read several _MissedReadingService_ and _Ob1G5StateMachine_ error messages that seem related to Bluetooth connection with the transmitter.
Which option can I test?
By the way, I already disabled the MIUI power management option but it doesn't affect the problem.

![1654679098361](https://user-images.githubusercontent.com/21258057/172578548-3b837d9a-d645-4d44-bb36-2fcb73d0aea7.jpg)
![1654679098370](https://user-images.githubusercontent.com/21258057/172578563-99c10b1e-98ad-4879-9975-2cb27f60e501.jpg)

**Comment by Navid200** [comment]
>  Eventually, I decided to switch back to the Dexcom app, and it still works, so it's an issue related to XDrip. 

Please explain what you mean by this.
Are you using the Dexcom app or xDrip?
You can NOT use both as collectors from the same transmitter at the same time.

The only way you can use both on the same phone is by using xDrip as a follower.

**Reply by dagord** [reply]
Hello.
I mean that I tested xDrip a few days, but since the connection was unreliable I returned to Dexcom app

**Reply by jamorham** [reply]
It should be possible to run the manufacturer app and xDrip side by side using the `Companion` data source present in the latest xDrip nightly. I would be interested to hear if this works and so OP might be a good candidate to try this.

Side note: If they were trying to use both at the same time with xDrip also acting as a collector that could cause the bluetooth errors pasted in the log above.

**Reply by Navid200** [reply]
> Hello. I mean that I tested xDrip a few days, but since the connection was unreliable I returned to Dexcom app

But, if you have returned to using Dexcom app, why are you showing xDrip logs and what do you expect us to do with xDrip?
I still don't understand what you are describing.

If you are using xDrip and want us to help you with it, why do you say you have switched to the Dexcom app.
Please read your post and make sure it presents the sequence of events correctly.

**Reply by dagord** [reply]
> If you are using xDrip and want us to help you with it, why do you say you have switched to the Dexcom app.
> Please read your post and make sure it presents the sequence of events correctly.

Well, I simply tested xDrip for a few days (removing the Dexcom app) but I experienced the problems described; the logs attached refer to these issues. 
So I switched back to Dexcom app as temporary solution; I'm asking for help since I would happily adopt xDrip, but in my case it's not a reliable tool.
I hope its clear: I'm not a native english speaker. :)

**Comment by Navid200** [comment]
If you want to use xDrip as the collector, you can try the latest Nightly because it has improvements to the collector:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.06.09

If you don't want to use xDrip as the collector.  But, still want to have it, you have several options.  
https://navid200.github.io/xDrip/docs/Follow_page.html  
Please read through the details and if anything is not clear, please let me know.

**Comment by dagord** [comment]
Just an update: apparently, I solved the issue.
When the old sensor expired, I uninstalled Dexcom app, rebooted the phone, installed xDrip again (lastest stable version as before) then started the new sensor.
Not sure what changed... But the connection seems stable as it never dropped since then.
Fingers crossed!
